substrate/frame/support/src/traits/randomness.rs

```diff
@@ -52,3 +56,85 @@ pub trait Randomness<Output, BlockNumber> {
 		Self::random(&[][..])
 	}
 }
+
+/// An object that is able to produce an arbitrary length slice using some initial set of bytes that
+/// will repeat over and over.
+pub struct RepeatingSlice<'a>(&'a [u8], usize);
```

Comment:
I wonder if you can just store a `std::iter::Cycle<std::slice::Iter<'a, u8>>`, but not sure how it work for empty input. `new` could return an error, since the returned repeater can never be read from.

Reply:
Iter is probably slower when it comes to copying the data over, because you copy byte by byte, instead of using `memcpy`